src/canvas/__tests__/eventData.test.ts

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable no-restricted-globals */
+import { IText, iMatrix } from '../../../fabric';
+import type { TMat2D } from '../../typedefs';
+import { Canvas } from '../Canvas';
+
+describe('Canvas event data', () => {
+  let canvas: Canvas;
+  let spy: jest.SpyInstance;
+
+  beforeEach(() => {
+    canvas = new Canvas(null);
+    spy = jest.spyOn(canvas, 'fire');
+  });
+
+  afterEach(() => {
+    return canvas.dispose();
+  });
+
+  test.each(
+    (
+      [
+        'mousedown',
+        'mouseenter',
+        'mouseleave',
+        'mousemove',
+        'mouseout',
+        'mouseover',
+        'dblclick',
+        'wheel',
+        'contextmenu',
+      ] as (keyof WindowEventMap)[]
+    )
+      .map(
+        (type) =>
+          [
+            [type, iMatrix],
+            [type, [2, Math.PI, 0, 0.5, 50, 50] as TMat2D],
```

Comment:
the choice for this generic matrix but with clientX and clientY matching the translation bring the absolutePointer always to 0,0 nullifing the effect of zoom and that unused skew/rotation extra parameter.

Let's have a matrix that cover our use cases, with 0 and 3 same number and 1 ad 2 set as zero, and the translation that is not the same as the mouse client coord.